src/test/java/com/spotify/docker/client/CompressedDirectoryTest.java

```diff
@@ -74,9 +78,31 @@ public void testFileWithIgnore() throws Exception {
         final String name = entry.getName();
         names.add(name);
       }
-      assertThat(names, containsInAnyOrder("Dockerfile", "bin/date.sh", "subdir2/keep.me",
-                                           "subdir2/do-not.ignore", "subdir3/do.keep",
-                                           ".dockerignore"));
+      assertThat(names, containsInAnyOrder("Dockerfile", "bin/", "bin/date.sh", "subdir2/",
+                                           "subdir2/keep.me", "subdir2/do-not.ignore",
+                                           "subdir3/do.keep", ".dockerignore"));
```

Comment:
That's an interesting one, since `subdir3` is in the [.dockerignore](https://github.com/spotify/docker-client/blob/master/src/test/resources/dockerDirectoryWithIgnore/.dockerignore) file it'll ignore that directory, but still include the `subdir3/do.keep` file. I think it's ok to handle it that way as the directory is still 'created' by the other file handler